code/game/machinery/ds13/stasis_station.dm

```diff
@@ -0,0 +1,62 @@
+/obj/machinery/stasis_station
+	name = "stasis recharge station"
+	desc = "This station is used to recharge stasis module."
+	icon = 'icons/obj/machines/ds13/stasis.dmi'
+	icon_state = "stasis_station"
+	anchored = 1
+	use_power = 0
+
+	var/busy = FALSE
+	var/recharging = FALSE
+
+/obj/machinery/stasis_station/Initialize()
+	. = ..()
+	update_icon()
+
+/obj/machinery/stasis_station/update_icon()
+	if(busy)
+		icon_state = "stasis_station_discharging"
+	else if(recharging)
+		icon_state = "stasis_recharging"
+	else
+		icon_state = "stasis_station"
+
+/obj/machinery/stasis_station/attack_hand(var/mob/living/carbon/human/user)
```

Comment:
don't pile this much code into attackby, have it call procs for doing specific recharging of a user/rig